### PR TITLE
fix(docs): fix most sphinx warnings

### PR DESCRIPTION
**Pull request by vwheeler63**
cc:  @kisvegabor

In PR #6542 Kevin made a number of valuable changes to .RST files the eliminate
a large number of Sphinx errors that he encountered.  Those were cherry picked
and reviewed and brought here.  On a test docs generation a few more cropped
up and they were handled here as well.

What it leaves UNhandled because it has something to do with `breathe` that I don't
quite get yet is a large number of warnings that look like this (all "Cannot find file"):
```
<tmpdir>\API/others/gridnav/lv_gridnav.rst:8: WARNING: doxygenfile: Cannot find file "lv_gridnav.h
```
and
```
<tmpdir>\widgets\led.rst:7: WARNING: duplicate label lv_led, other instance in <tmpdir>\API/widgets/led/lv_led.rst
```
which appears to be caused by an error in whatever preps the API .RST files for Sphinx.  In the above example, I believe either this file: `<tmpdir>\API/widgets/led/lv_led.rst` or this file `<tmpdir>\widgets\led.rst` is in error, not sure which.

I am hoping Kevin ( @kdschlosser ) might know what to do with these.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.  n/a
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.  n/a
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.  n/a
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).  n/a
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).  n/a
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.  (is ready)
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.  Done.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by vwheeler63 — COMMENTED** [review]
I have reviewed each file and can assert that it is as it needs to be for correct formatting and eliminating Sphinx errors.  The `lv_conf_template.h` and `lv_conf_internal.h` are minor formatting updates that got overlooked previously.

**Comment by kdschlosser** [comment]
the duplicate labels are happening because there is a duplicate reference. It is a crappy error/warning that sphinx uses and it really needs to be changed upstream in sphinx.


If you look at the hard coded `.rst` files in the `docs` directory you will see `.. _some-reference:` above the header. That is a "label" It is actually an anchor for a reference. In the example you have posted above the problem needs to be fixed in the generation code for the API...

https://github.com/lvgl/lvgl/blob/0227f7dec5436c3fa59f86836c955b14aa1c91fc/docs/doc_builder.py#L1099

needs to be changed to 

`f.write('.. _{0}_h:'.format(name))`

and https://github.com/lvgl/lvgl/blob/0227f7dec5436c3fa59f86836c955b14aa1c91fc/docs/doc_builder.py#L1334 needs to be changed to

`':ref:`{0}_h`\n'.format(inc)`

That will stop the label collisions.

There has been a massive change to the API naming and that includes filenames. If the hard coded `.rst` files have specific information that aligns with a specific piece of the API the filename omf the `.rst` file MUST match the header filename that the information is about. This is what links the hard coded `.rst` file with the proper API section in the documentation.

**Comment by vwheeler63** [comment]
@FASTSHIFT 

> changed the title Fix Most Sphinx Warnings fix(docs): fix most sphinx warnings

Got it.  😉

**Comment by vwheeler63** [comment]
@kisvegabor Ready with cherry-picked PR if you hadn't seen this already.

**Comment by vwheeler63** [comment]
> The duplicate labels are happening because...

Thank you for the tips!  It works!

> If the hard coded `.rst` files have specific information that aligns with a specific piece of the API the filename of the `.rst` file MUST match the header filename that the information is about.

Got it.  Handled.

> The problem with the not finding files is the reason why I went looking for a breathe replacement.

Let's move this to #6924

**Review by kisvegabor — DISMISSED** [review]
Thank you Victor and Kevin.
Having these in place would make further work on the docs simpler and lighter.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# Documentation
+
+----------------------------------------------
+
+## Building
+
+Building the documentation is pretty easy to do but it does have some requirements
+That have to be filled prior to building them.
```
Comment (liamHowatt) [inline docs/README.md]:
```suggestion
Building the documentation is pretty easy to do but it does have some requirements
that have to be filled prior to building them.
```

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+    pip install -r requirements.txt
+
+
+Once you have all of the requirements installed you are ready to build them.
+To build the documentation use the following command.
+
+    python build.py skip_latex clean
+
+You may have to use the following command if you are un a Unix like OS
```
Comment (liamHowatt) [inline docs/README.md]:
```suggestion
You may have to use the following command if you are on a Unix like OS
```

**Comment by vwheeler63** [comment]
> Thank you Victor and Kevin. Having these in place would make further work on the docs simpler and lighter.

I think we can eliminate all the `sphinx` warnings!  As I type this, there is only one type left, and I know there was a point a couple of months ago when I was generating documentation without this type of error, so I think it is possible.

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+### index.rst files
+
+--------------------------
+
+If you create a new directory you MUST have an `index.rst` file in that directory
+and that index file needs to be pointed to in the `index.rst` file that is located
+in the parent directory.
+
+Lets take a look at the `index.rst` file that is located in the `docs/layouts` directory.
```
Comment (liamHowatt) [inline docs/README.md]:
```suggestion
Let's take a look at the `index.rst` file that is located in the `docs/layouts` directory.
```

File: docs/integration/chip/nxp.rst [hunk 4]
```diff
@@ -50,30 +50,31 @@ Basic initialization:
...
 
-.. code:: c
+.. code-block:: c
 
-  #if LV_USE_PXP_DRAW_THREAD
-    lv_thread_init(&draw_pxp_unit->thread, LV_THREAD_PRIO_HIGH, _pxp_render_thread_cb, 2 * 1024, draw_pxp_unit);
-  #endif
+    #if LV_USE_OS
+        lv_thread_init(&draw_pxp_unit->thread, LV_THREAD_PRIO_HIGH, _pxp_render_thread_cb, 2 * 1024, draw_pxp_unit);
+    #endif
```
Comment (liamHowatt) [inline docs/integration/chip/nxp.rst]: How come this was changed? https://github.com/lvgl/lvgl/blob/9db0ee3d02a113058c8296d68eee98baa7967d6f/src/draw/nxp/pxp/lv_draw_pxp.c#L94-L96

**Review by liamHowatt — COMMENTED** [review]
Thanks!

File: docs/README.md (hunk 3 above)
Comment (kdschlosser) [reply docs/README.md]: get so used to the spell correcting that takes place on cell phones it carries over into the land of programming especially when writing the documentation. No automatic spelling correction when writing the documentation!!!

Comment (kisvegabor) [reply docs/README.md]: I'm not using it yet, but maybe copilot can hep with that too.

File: docs/integration/chip/nxp.rst (hunk 4 above)
Comment (vwheeler63) [reply docs/integration/chip/nxp.rst]: @liamHowatt Oh my, my oversight.  That was outdated code from a cherry-pick that I overlooked.  It was only meant to include the indentation change for consistency.  Fixed in current commit.

Comment (vwheeler63) [reply docs/integration/chip/nxp.rst]: @kisvegabor @liamHowatt   Ready for review again.  FYI, Liam, I didn't proof-read these for grammar and clarity yet.  That will be done in the near future.

**Review by kisvegabor — DISMISSED** [review]

File: docs/integration/chip/nxp.rst [hunk 5]
```diff
@@ -263,11 +269,11 @@ created with the additional callbacks:
...
 
-.. code:: c
+.. code-block:: c
 
-  #if LV_USE_VGLITE_DRAW_THREAD
-    lv_thread_init(&draw_vglite_unit->thread, LV_THREAD_PRIO_HIGH, _vglite_render_thread_cb, 2 * 1024, draw_vglite_unit);
-  #endif
+    #if LV_USE_OS
+        lv_thread_init(&draw_vglite_unit->thread, LV_THREAD_PRIO_HIGH, _vglite_render_thread_cb, 2 * 1024, draw_vglite_unit);
+    #endif
```
Comment (liamHowatt) [inline docs/integration/chip/nxp.rst]: One more. Sorry.

**Review by liamHowatt — DISMISSED** [review]

**Comment by liamHowatt** [comment]
Thanks! Looks like there are merge conflicts.

File: docs/integration/chip/nxp.rst (hunk 5 above)
Comment (vwheeler63) [reply docs/integration/chip/nxp.rst]: > One more. Sorry.

cc:  @kisvegabor 

Handled and re-submitted.  Thank you!  Yipes.  I see the conflicts.  I'm on it.

**Comment by vwheeler63** [comment]
> Thanks! Looks like there are merge conflicts.

I'm on it.

**Comment by vwheeler63** [comment]
@liamHowatt @kisvegabor 

Conflicts handled and re-submitted.  FYI, the underline lengths previously discussed also match the titles in this one.

**Comment by kisvegabor** [comment]
There is a new conflict. 🙁

**Comment by vwheeler63** [comment]
> There is a new conflict. 🙁

No problem!  Conflict fixed & re-submitted.

**Review by liamHowatt — APPROVED** [review]
Thank you. I did not check thoroughly. I will assume the conflict resolution was ok :slightly_smiling_face:

**Comment by vwheeler63** [comment]
> Thank you. I did not check thoroughly. I will assume the conflict resolution was ok 🙂

Indeed:  mostly spaces at the end of 7 lines, which I had eliminated in a previously-merged commit, and not in this one.

**Review by kisvegabor — APPROVED** [review]

**Comment by kisvegabor** [comment]
> The long dashed lines get changed into a horizontal line or hrule

I know, but I think we don't need these custom decorations, but for consistency we should rely on the markdown viewer's style.

**Comment by vwheeler63** [comment]
> > The long dashed lines get changed into a horizontal line or hrule
> 
> I know, but I think we don't need these custom decorations, but for consistency we should rely on the markdown viewer's style.

Done!  PR #6992

**Comment by kdschlosser** [comment]
> I know, but I think we don't need these custom decorations, but for consistency we should rely on the markdown viewer's style.

markdown viewer or ReST viewer?

**Comment by vwheeler63** [comment]
> > I know, but I think we don't need these custom decorations, but for consistency we should rely on the markdown viewer's style.
> 
> markdown viewer or ReST viewer?

Markdown.

**Pull request merged**